packages/manager/apps/hpc-vmware-managed-vcd/src/pages/dashboard/datacentre/compute-order/DatacentreComputeOrder.page.tsx

```diff
@@ -0,0 +1,168 @@
+import { subRoutes } from '@/routes/routes.constant';
+import { IVdcOrderableVhostPriced } from '@/types/vcd-vdc-orderable-resource.interface';
+import { validateComputeQuantity } from '@/utils/formValidation';
+import { useVdcOrderableResource } from '@/data/hooks/useOrderableResource';
+import { useVcdCatalog } from '@/data/hooks/useVcdCatalog';
+import { getPricedOrderableVhostList } from '@/utils/getPricedOrderableResource';
+import useVcdOrder from '@/data/hooks/useVcdOrder';
+
+export default function ComputeOrderPage() {
+  const { t } = useTranslation('hpc-vmware-managed-vcd/datacentres/compute');
+  const { id, vdcId } = useParams();
+  const navigate = useNavigate();
+  const [selectedVhost, setSelectedVhost] = useState<string>(null);
+  const [selectedQuantity, setSelectedQuantity] = useState<number>(1);
+  const { data, isLoading, isError } = useVdcOrderableResource(id, vdcId);
+  const {
+    data: catalog,
+    isLoading: isLoadingCatalog,
+    isError: isCatalogError,
+  } = useVcdCatalog(id);
+  const { redirectToOrder } = useVcdOrder({
+    serviceName: id,
+    planCode: selectedVhost,
+    quantity: selectedQuantity,
+  });
+
+  const isValidQuantity = validateComputeQuantity(selectedQuantity);
+  const orderableVhostList = getPricedOrderableVhostList({
+    vhostList: data?.data?.compute,
+    catalog: catalog?.data,
+  });
+
+  useEffect(() => {
+    if (orderableVhostList?.length && !selectedVhost) {
+      setSelectedVhost(orderableVhostList[0].profile);
+    }
+  }, [selectedVhost, orderableVhostList]);
+
+  if (isLoading || isLoadingCatalog) return <Loading />;
+  if (isError || isCatalogError || !orderableVhostList?.length) {
+    return (
+      <ErrorBanner
+        error={{
+          status: 500,
+          data: {
+            message: t('managed_vcd_vdc_compute_order_vhost_unavailable'),
+          },
+        }}
+      />
+    );
+  }
+
+  const columns: DatagridColumn<IVdcOrderableVhostPriced>[] = [
+    {
+      id: 'select',
+      cell: (vhost) => (
+        <ComputeOrderSelectCell
+          vHost={vhost}
+          selectedVhost={selectedVhost}
+          setSelectedVhost={setSelectedVhost}
+        />
+      ),
+      label: '',
+      isSortable: false,
+    },
+    {
+      id: 'vhost',
+      cell: ComputeOrderVhostCell,
+      label: t('managed_vcd_vdc_compute_order_vhost'),
+      isSortable: false,
+    },
+    {
+      id: 'cpuSpeed',
+      cell: ComputeOrderCpuSpeedCell,
+      label: t('managed_vcd_vdc_compute_order_cpu_speed'),
+      isSortable: false,
+    },
+    {
+      id: 'ram',
+      cell: ComputeOrderRamCell,
+      label: t('managed_vcd_vdc_compute_order_ram'),
+      isSortable: false,
+    },
+    {
+      id: 'cpuCount',
+      cell: ComputeOrderCpuCountCell,
+      label: t('managed_vcd_vdc_compute_order_vcpu_count'),
+      isSortable: false,
+    },
+    {
+      id: 'price',
+      cell: ComputeOrderPriceCell,
+      label: t('managed_vcd_vdc_compute_order_price'),
```

Comment:
so price could be sorted ?

Reply:
No, it's a mistake: has to be sorted by price by default